app/merginapi.cpp

```diff
@@ -379,43 +378,44 @@ void MerginApi::handleDataStream(QNetworkReply* r, QString projectDir)
             if (!activeFilePath.isEmpty()) {
                 QRegularExpressionMatch endMatch = endPattern.match(data);
                 int tillIndex = data.indexOf(endMatch.captured(0));
-                saveFile(data.left(tillIndex), activeFilePath, true);
-                activeFilePath = "";
+                QFile activeFile(activeFilePath);
+                saveFile(data.left(tillIndex), activeFile, true);
             }
             return;
         }
 
         data = data.append(chunk);
-        dataString = QString::fromStdString(data.toStdString());
+        dataString = QString::fromLatin1(data.data(), data.size());
         QRegularExpressionMatch boundaryMatch = boundaryPattern.match(dataString);
 
         while (boundaryMatch.hasMatch()) {
             if (!activeFilePath.isEmpty()) {
                 int tillIndex = data.indexOf(boundaryMatch.captured(0));
-                saveFile(data.left(tillIndex), activeFilePath, true);
-                activeFilePath = "";
+                QFile activeFile(activeFilePath);
```

Comment:
This defines "activeFile" variable in the local block and thus shadows the other "activeFile" from the outer scope!